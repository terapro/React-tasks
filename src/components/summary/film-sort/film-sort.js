import React from 'react';


import {Button} from 'src/components/common/button/button.js'
import {Label} from 'src/components/common/label/label.js'

import PropTypes from 'prop-types';


const FilmSort = (props) => {
  const {sortItems, changeSortItemCallBack} = props;
  if (sortItems) {
    return (
      <div className='film-sort'>
        <Label content='Sort by '/>
          {
            sortItems.list.map(
              (item) => (
                <Button key={item}
                        btnTranspRed  = {item === sortItems.active}
                        btnTransp  = {item !== sortItems.active}
                        btnClick = {changeSortItemCallBack}
                        btnBold
                        content = {item}
                        id = {item}
                />
              )
            )
          }
      </div>
    );
  } else return '';

};

FilmSort.propTypes = {
  changeSortItemCallBack: PropTypes.func
};
FilmSort.defaultProps = {
  changeSortItemCallBack: ()=>{}
};





export {FilmSort};