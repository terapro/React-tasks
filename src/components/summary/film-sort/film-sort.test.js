import React from 'react';
import {FilmSort} from 'src/components/summary/film-sort/film-sort';
import {shallow} from 'enzyme';

const mockFunc = jest.fn();
const mockSortParameters = {
  parameters: ['release date', 'rating', 'my rating'],
  chosenParameter: 'release date'
};

describe('<FilmSort />', () => {
  describe('Render', () => {
    it('should render the component', () => {
      const wrapper = shallow(<FilmSort onParameterClick={mockFunc}
                                        sortParameters={mockSortParameters}
      />);
      expect(wrapper).toMatchSnapshot();
    });
    it('should render the component with default props', () => {
      const wrapper = shallow(<FilmSort sortParameters={mockSortParameters} />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});