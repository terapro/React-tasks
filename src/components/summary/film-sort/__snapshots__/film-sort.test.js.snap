// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilmSort /> Render should render the component 1`] = `
<div
  className="film-sort"
>
  <Label
    content="Sort by "
    labelBold={false}
    labelUC={false}
    labelWhite={false}
  />
  <Button
    btnBig={false}
    btnBold={true}
    btnClick={[MockFunction]}
    btnGrey={false}
    btnRed={false}
    btnSm={false}
    btnTransp={false}
    btnTranspRed={true}
    btnUC={false}
    btnWhite={false}
    content="release date"
    id="release date"
    key="release date"
  />
  <Button
    btnBig={false}
    btnBold={true}
    btnClick={[MockFunction]}
    btnGrey={false}
    btnRed={false}
    btnSm={false}
    btnTransp={true}
    btnTranspRed={false}
    btnUC={false}
    btnWhite={false}
    content="rating"
    id="rating"
    key="rating"
  />
  <Button
    btnBig={false}
    btnBold={true}
    btnClick={[MockFunction]}
    btnGrey={false}
    btnRed={false}
    btnSm={false}
    btnTransp={true}
    btnTranspRed={false}
    btnUC={false}
    btnWhite={false}
    content="my rating"
    id="my rating"
    key="my rating"
  />
</div>
`;

exports[`<FilmSort /> Render should render the component with default props 1`] = `
<div
  className="film-sort"
>
  <Label
    content="Sort by "
    labelBold={false}
    labelUC={false}
    labelWhite={false}
  />
  <Button
    btnBig={false}
    btnBold={true}
    btnClick={[Function]}
    btnGrey={false}
    btnRed={false}
    btnSm={false}
    btnTransp={false}
    btnTranspRed={true}
    btnUC={false}
    btnWhite={false}
    content="release date"
    id="release date"
    key="release date"
  />
  <Button
    btnBig={false}
    btnBold={true}
    btnClick={[Function]}
    btnGrey={false}
    btnRed={false}
    btnSm={false}
    btnTransp={true}
    btnTranspRed={false}
    btnUC={false}
    btnWhite={false}
    content="rating"
    id="rating"
    key="rating"
  />
  <Button
    btnBig={false}
    btnBold={true}
    btnClick={[Function]}
    btnGrey={false}
    btnRed={false}
    btnSm={false}
    btnTransp={true}
    btnTranspRed={false}
    btnUC={false}
    btnWhite={false}
    content="my rating"
    id="my rating"
    key="my rating"
  />
</div>
`;
